lib/functions/rootfs/distro-agnostic.sh

```diff
@@ -361,75 +362,40 @@ function install_distribution_agnostic() {
-		install_deb_chroot "${DEB_STORAGE}/${BSP_CLI_PACKAGE_FULLNAME}.deb"
-	else
-		install_deb_chroot "${CHOSEN_ROOTFS}" "remote" # @TODO: rpardini: err.... what?
-	fi
+	install_deb_chroot "${DEB_STORAGE}/${image_artifacts_debs["armbian-bsp-cli"]}"
 
 	# install armbian-desktop
-	if [[ "${REPOSITORY_INSTALL}" != *armbian-desktop* ]]; then
-		if [[ $BUILD_DESKTOP == yes ]]; then
-			install_deb_chroot "${DEB_STORAGE}/${RELEASE}/${CHOSEN_DESKTOP}_${REVISION}_all.deb"
-			install_deb_chroot "${DEB_STORAGE}/${RELEASE}/${BSP_DESKTOP_PACKAGE_FULLNAME}.deb"
-			# install display manager and PACKAGE_LIST_DESKTOP_FULL packages if enabled per board
-			desktop_postinstall
-		fi
-	else
-		if [[ $BUILD_DESKTOP == yes ]]; then
-			install_deb_chroot "${CHOSEN_DESKTOP}" "remote"
-			# install display manager and PACKAGE_LIST_DESKTOP_FULL packages if enabled per board
-			desktop_postinstall
-		fi
+	if [[ $BUILD_DESKTOP == yes ]]; then
+		install_deb_chroot "${DEB_STORAGE}/${image_artifacts_debs["armbian-desktop"]}"
+		install_deb_chroot "${DEB_STORAGE}/${image_artifacts_debs["armbian-bsp-desktop"]}"
+		# install display manager and PACKAGE_LIST_DESKTOP_FULL packages if enabled per board
+		desktop_postinstall
 	fi
 
 	# install armbian-config
 	if [[ "${PACKAGE_LIST_RM}" != *armbian-config* ]]; then
-		if [[ "${REPOSITORY_INSTALL}" != *armbian-config* ]]; then
-			if [[ $BUILD_MINIMAL != yes ]]; then
-				install_deb_chroot "${DEB_STORAGE}/armbian-config_${REVISION}_all.deb"
-			fi
-		else
-			if [[ $BUILD_MINIMAL != yes ]]; then
-				install_deb_chroot "armbian-config" "remote"
-			fi
+		if [[ $BUILD_MINIMAL != yes ]]; then
+			install_deb_chroot "${DEB_STORAGE}/${image_artifacts_debs["armbian-config"]}"
 		fi
 	fi
 
 	# install armbian-zsh
 	if [[ "${PACKAGE_LIST_RM}" != *armbian-zsh* ]]; then
-		if [[ "${REPOSITORY_INSTALL}" != *armbian-zsh* ]]; then
-			if [[ $BUILD_MINIMAL != yes ]]; then
-				install_deb_chroot "${DEB_STORAGE}/armbian-zsh_${REVISION}_all.deb"
-			fi
-		else
-			if [[ $BUILD_MINIMAL != yes ]]; then
-				install_deb_chroot "armbian-zsh" "remote"
-			fi
+		if [[ $BUILD_MINIMAL != yes ]]; then
+			install_deb_chroot "${DEB_STORAGE}/${image_artifacts_debs["armbian-config"]}"
 		fi
 	fi
 
-	# install plymouth-theme-armbian
+	# install armbian-plymouth-theme
 	if [[ $PLYMOUTH == yes ]]; then
-		if [[ "${REPOSITORY_INSTALL}" != *plymouth-theme-armbian* ]]; then
-			install_deb_chroot "${DEB_STORAGE}/armbian-plymouth-theme_${REVISION}_all.deb"
-		else
-			install_deb_chroot "armbian-plymouth-theme" "remote"
-		fi
+		install_deb_chroot "${DEB_STORAGE}/${image_artifacts_debs["armbian-plymouth-theme"]}"
 	fi
 
 	# install wireguard tools
 	if [[ $WIREGUARD == yes ]]; then
 		install_deb_chroot "wireguard-tools" "remote"
 	fi
 
-	# freeze armbian packages
-	if [[ $BSPFREEZE == yes ]]; then
-		display_alert "Freezing Armbian packages" "$BOARD" "info"
-		# @TODO: rpardini: this will probably fail if one or more packages are not installed
```

Comment:
This is important feature. Can't be removed. And works even some packages are missing:
```
espressobin-ultra:~:# apt-mark hold armbian-config dsds
armbian-config set on hold.
E: Unable to locate package dsds
espressobin-ultra:~:# echo $?
100

```

Reply:
That feature didn't work. Ignoring the error is not solution. Need to keep list of actually installed pkgs and hold the ones that are actually installed. Ignoring error is a bad idea

Reply:
Brought it back with a explicit break and a TODO waiting just for you... ;-)